common/models/zod-schemas.ts

```diff
@@ -55,3 +56,40 @@ export const OttApiRequestAccountRecoveryVerifySchema = z.object({
 	verifyKey: z.string(),
 	newPassword: z.string(),
 });
+
+const CategorySchema = z.enum([
+	"sponsor",
+	"intro",
+	"outro",
+	"interaction",
+	"selfpromo",
+	"music_offtopic",
+	"preview",
+]);
+
+const GrantSchema = z.tuple([
+	z.object({
+		role: z.nativeEnum(Role),
+	}),
+	z.object({
+		grantMask: z.number(),
+	}),
+]);
+
+const RoomSettingsSchema = z.object({
+	title: z.string().max(255).optional(),
+	description: z.string().optional(),
+	visibility: z.nativeEnum(Visibility).optional(),
+	queueMode: z.nativeEnum(QueueMode).optional(),
+	grants: z.array(GrantSchema).optional(),
+	autoSkipSegmentCategories: z.array(CategorySchema).optional(),
+	restoreQueueBehavior: z.nativeEnum(BehaviorOption).optional(),
+	enableVoteSkip: z.boolean().optional(),
+});
+
+const ClaimSchema = z.object({
+	claim: z.boolean().optional(),
+	...RoomSettingsSchema.shape,
+});
```

Comment:
I meant just the object with `claim`. It shouldn't include room settings.
```suggestion
const ClaimSchema = z.object({
	claim: z.boolean().optional(),
});
```